packages/amplify-category-function/provider-utils/awscloudformation/index.js

```diff
@@ -199,4 +199,18 @@ async function invoke(context, category, service, resourceName) {
   });
 }
 
-module.exports = { addResource, invoke };
+function migrateResource(projectPath, service, resourceName) {
+  serviceMetadata = JSON.parse(fs.readFileSync(`${__dirname}/../supported-services.json`))[service];
+  const { serviceWalkthroughFilename } = serviceMetadata;
+  const serviceWalkthroughSrc = `${__dirname}/service-walkthroughs/${serviceWalkthroughFilename}`;
+  const { migrate } = require(serviceWalkthroughSrc);
+
+  if (!migrate) {
+    console.log(`No migration required for ${resourceName}`);
+  }
+
+  return migrate(projectPath, resourceName);
```

Comment:
if migrate can be undefined or null, then this line needs to be combined with the previous if block, call migrate only if it's defined.